shift_detector/precalculations/EmbeddingDistancePrecalculation.py

```diff
@@ -0,0 +1,67 @@
+from shift_detector.precalculations.TextEmbeddingPrecalculation import TextEmbeddingPrecalculation
+from shift_detector.precalculations.Store import Store
+from shift_detector.precalculations.Precalculation import Precalculation
+from datawig.utils import random_split
+from math import sqrt
+
+
+class EmbeddingDistancePrecalculation(Precalculation):
+
+    def __init__(self, model=None, trained_model=None):
+        self.model = model
+        self.trained_model = trained_model
+
+    def __eq__(self, other):
+        return self.model == other.model \
+               and self.trained_model == other.trained_model
+
+    def __hash__(self):
+        return hash((self.model, self.trained_model))
+
+    @staticmethod
+    def calculate_distance(vector1, vector2):
```

Comment:
This is standard stuff and should be already implemented in one of our libraries.